lib/maplibre.ex

```diff
@@ -526,4 +595,78 @@ defmodule MapLibre do
   defp to_style(%{}), do: %{"version" => 8}
   defp to_style(style) when is_map(style), do: style
   defp to_style(style), do: Jason.decode!(style)
+
+  defp geometry_from_table(data, spec, []) do
+    geometries =
+      data
+      |> points(spec)
+      |> Enum.map(&%Geo.Point{coordinates: &1})
+
+    Geo.JSON.encode!(%Geo.GeometryCollection{geometries: geometries}, feature: true)
+  end
+
+  defp geometry_from_table(data, spec, properties) do
+    points = points(data, spec)
+    properties = properties(data, properties)
+    geometries = Enum.zip(points, properties)
+
+    geometries =
+      for {point, props} <- geometries, do: %Geo.Point{coordinates: point, properties: props}
+
+    Geo.JSON.encode!(%Geo.GeometryCollection{geometries: geometries}, feature: true)
+  end
+
+  def points(data, {format, [lng, lat]}) do
+    validate_columns!(data, [lng, lat])
+
+    data
+    |> Table.to_columns(only: [lng, lat])
+    |> then(&Enum.zip(&1[lng], &1[lat]))
+    |> Enum.map(&parse_coordinates(&1, format))
```

Comment:
You can use zip_with (from Elixir v1.12) and avoid one additional traversal:

```suggestion
    table = Table.to_columns(data, only: [lng, lat])
    Enum.zip_with(table[lng], table[lat], &parse_coordinates(&1, &2, format))
```